cupy/cuda/memory.pyx

```diff
@@ -581,8 +580,7 @@ cpdef get_allocator():
         return allocator
 
 
-@contextlib.contextmanager
-def using_allocator(allocator=None):
+def _using_allocator(allocator=None):
```

Comment:
Why not just move this function to `cuda/__init__.py`?

Reply:
I tried that, but it relies on implementation details only accessible in that module. Eventually I gave up and left it there adding a wrapper in `cuda/__init__.py`. After all my main interest was just getting it in the docs 😉

That said, if you are able to find a way to fix this, that would be great. 😄

Reply:
Thank you for a comment!
I sent a fix as #3094.
I'm not 100% sure this is version is better, but please take a look.